ganttproject/src/net/sourceforge/ganttproject/ResourceTreeTable.java

```diff
@@ -123,12 +131,93 @@ protected Chart getChart() {
     return myUiFacade.getResourceChart();
   }
 
+  private static class AscendingNameComparator implements Comparator<HumanResource> {
+    @Override
+    public int compare(HumanResource o1, HumanResource o2) {
+      return o1.getName().compareTo(o2.getName());
+    }
+  }
+
+  private static class DescendingNameComparator implements Comparator<HumanResource> {
+    @Override
+    public int compare(HumanResource o1, HumanResource o2) {
+      return o2.getName().compareTo(o1.getName());
+    }
+  }
+
+
   /** Initialize the treetable. Addition of various listeners, tree's icons. */
   @Override
   protected void doInit() {
     super.doInit();
     myResourceTreeModel.updateResources();
+    getTable().getModel().addTableModelListener(new ModelListener());
     getVerticalScrollBar().addAdjustmentListener(new VscrollAdjustmentListener(myUiFacade.getResourceChart(), false));
+    getTableHeader().addMouseListener(new MouseAdapter() {
+      @Override
+      public void mouseClicked(MouseEvent mouseEvent) {
+        int index = getTable().columnAtPoint(mouseEvent.getPoint());
```

Comment:
I understand this was copied from `GanttTreeTable`, right? Can you please refactor it somehow, so that the logic at lines 159-173 was common to both classes? One option is to extract this code to a method in `GPTreeTableBase` and call it from both listeners in the extended classes. Another option is to move the listener itself and this code to `GPTreeTableBase` and call an abstract method once execution passes all these checks -- and let abstract method implementations do all the remaining work.